api/src/main/java/uk/gov/dhsc/htbhf/hmrc/service/EligibilityService.java

```diff
@@ -25,15 +25,18 @@
     private final HouseholdRepository repository;
     private final HouseholdVerifier householdVerifier;
     private final RestTemplate restTemplate;
+    private final EligibilityResponseFactory eligibilityResponseFactory;
 
     public EligibilityService(@Value("${hmrc.base-uri}") String baseUri,
                               HouseholdRepository repository,
                               HouseholdVerifier householdVerifier,
-                              RestTemplate restTemplate) {
+                              RestTemplate restTemplate,
+                              EligibilityResponseFactory eligibilityResponseFactory) {
```

Comment:
Yes, I definitely preferred it the previous way using a static method call rather than plumbing in the `EligibilityResponseFactory`.

Reply:
It felt like there was too much logic to not make it a dependency. The key for me was when the EligibilityServiceTest was failing because of an issue with the factory.